pagefaultgames/pokerogue
# [Localization] Partial Localization of Quick Claw Modifier

## What are the changes?
The message triggered by the Quick Claw modifier is hard coded right now. To address this, this PR starts by providing an english and german localization for the effect and uses the i18n plugin to show the correct translation to the user.
Additionally, the shop item for the quick claw got its correct german name (was just the english name before).

## Why am I doing these changes?
Having a hard coded english message might pose a problem to players with insufficient english skills as they might not understand the message. This could lead to frustration if, for example, the enemy pokemon uses their quick claw and the player does not understand why the were outsped.

## What did change?
The message text indicating that a pokemon just used its quick claw ("[PokemonName] used its Quick Claw to move faster!") was hard coded like this.
This PR starts by adding german and english localization for this text using the i18n plugin. Additionally, the german name for the quick claw item was added (was just the english name before).
**Unfortunately, I was not able to find any resources online on what this message is like in other languages in the main games, so I left out the localization for other languages for now.**

### Screenshots/Videos
Old version (german language but english message):
![image](https://github.com/pagefaultgames/pokerogue/assets/67913362/4eb76c12-54bf-413e-be86-73ee5feaf068)

New version in german:
![image](https://github.com/pagefaultgames/pokerogue/assets/67913362/6e187021-0c0e-469e-8d76-a3d99de06a3c)

New english version in english:
![image](https://github.com/pagefaultgames/pokerogue/assets/67913362/7abdda50-0859-4677-924f-95d13a7d709c)

New german name of item in shop:
![image](https://github.com/pagefaultgames/pokerogue/assets/67913362/ceee8ea1-f4c0-4d4e-8c22-3419be40f6d0)

**No I did not change the maximum held item count for the quick claw despite what the images suggest. That was just via the override file for simpler testing.**

## How to test the changes?
1. Add `{name: "QUICK_CLAW", count: 10}` to the `STARTING_HELD_ITEMS_OVERRIDE` and `"QUICK_CLAW"` to `ITEM_REWARD_OVERRIDE` in the `src/overrides.ts` file.
2. Start a new run.
3. Click an attack and check the displayed message.
4. Defeat a pokemon and check the first reward item's name.
5. Repeat in german and english.

## Checklist
